IPs/RSKIP502.md

```diff
@@ -0,0 +1,175 @@
+
+### 4. Union Bridge locking cap
+
+A locking cap will be implemented to control the maximum amount of RBTC that can be transferred to the Union Bridge. This cap helps mitigate risks by preventing excessive transfers beyond the system's control limits.
+
+In order to integrate the locking cap logic with the Union Bridge, new methods need to be added to the existing Bridge contract to enable querying and updating the locking cap. Based on RSKIP134 [3], which provides a standard mechanism for managing caps in smart contracts, new methods are defined.
+
+#### 4.1 Method to get the current locking cap
+
+A method will be added to allow anyone to query the current locking cap. This ensures transparency regarding the maximum allowable amount that can be transferred to the Union Bridge.
+
+**Method signature:**
+
+```
+function getUnionBridgeLockingCap() public view returns (uint256);
+```
+
+#### 4.2 Method to adjust the locking cap
+
+This method will allow authorized accounts to adjust the locking cap. The cap can only be increased, and the increase should be subject to a predefined maximum (e.g., doubling the current cap). The purpose of the adjustment method is to ensure that the cap can scale with demand, but only in a controlled way.
+
+**Method signature:**
+
+```
+function increaseUnionBridgeLockingCap(uint256 newCap) public returns (int256);
+```
+
+**Response codes:**
+
+- 0: Success – The locking cap has been adjusted successfully.
+-1: Unauthorized caller – Only authorized entities can adjust the locking cap.
+-2: Invalid value – The specified cap value is invalid (e.g., less than the current cap or excessive).
+- -10: Generic error – A non-specific error occurred while processing the request.
+
+### 5. Method for Transferring RBTC to the Union Bridge
+
+A new method will be added to the Bridge contract to allow the transfer of RBTC to the Union Bridge. The method will have the following restrictions:
+
+- Only callable by the Union Bridge contract.
+- The total requested amount (current request + previously requested amounts) must not exceed the current locking cap.
+
+**Method Signature:** 
+```
+function requestRBTC(uint256 amountInWeis) public returns
+```
+
+An event will be emitted each time this method is invoked, to provide transparency about the transfer.
+
+```
+event rbtc_requested(address indexed requester, uint256 amountInWeis);
+```
+
+### 6. Tracking transfers to the Union Bridge
+
+A new storage entry will be used to track the total amount of RBTC that has been transferred to the Union Bridge. This is essential to ensure that the locking cap is not exceeded.
+
+**Storage entry name:** weisTransferredToUnionBridge
+
+Tracks the total amount of RBTC (expressed in weis) transferred to the Union Bridge.
+
+### 7. Receiving funds back from the Union Bridge
+
+The Union Bridge contract will be capable of sending funds back to the PowPeg. When this happens, the tracking entry for the transferred amount will need to be updated.
```

Comment:
add: the bridge will not allow weisTransferredToUnionBridge to go negative, which could only happen if the union bridge is malfunctioning. If this happens, the method will fail with an error code and  the bridge will automatically enter the pause mode.
The best path forward in this case we can discuss.